pythonPackages.numba: 0.51.0 -> 0.52.0

veprbl opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

veprbl committed:
pythonPackages.llvmlite: 0.34.0 -> 0.35.0

veprbl committed:
pythonPackages.numba: 0.51.0 -> 0.52.0

SuperSandro2000 commented on pkgs/development/python-modules/numba/default.nix:
```diff
@@ -11,14 +11,14 @@
+  version = "0.52.0";
   pname = "numba";
   # uses f-strings
   disabled = pythonOlder "3.6";
```
```
builder for '/nix/store/kdjihdn27rcqyadj6j7fp6rlyfmi6x67-python3.9-numba-0.52.0.drv' failed with exit code 1; last 10 log lines:                                                              building                                                                                                                                                                                    Executing setuptoolsBuildPhase                                                                                                                                                              Traceback (most recent call last):                                                                                                                                                            File "/build/numba-0.52.0/nix_run_setup", line 8, in <module>                                                                                                                                 exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))                                                                                File "setup.py", line 50, in <module>                                                                                                                                                         _guard_py_ver()                                                                                                                                                                           File "setup.py", line 47, in _guard_py_ver                                                                                                                                                    raise RuntimeError(msg.format(cur_py, min_py, max_py))                                                                                                                                  RuntimeError: Cannot install on Python version 3.9.2; only versions >=3.6,<3.9 are supported.
```

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114572` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
12 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.arviz</li>
<li>python37Packages.dftfit</li>
<li>python37Packages.pygbm</li>
<li>python37Packages.pymc3</li>
<li>python38Packages.arviz</li>
<li>python38Packages.dftfit</li>
<li>python38Packages.pygbm</li>
<li>python38Packages.pymc3</li>
<li>python39Packages.arviz</li>
<li>python39Packages.dftfit</li>
<li>python39Packages.pygbm</li>
<li>python39Packages.pymc3</li>
</ul>
</details>
<details>

  <summary>8 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python37Packages.shap: <a href="https://termbin.com/ncgk">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/6ecaebc504cd0dec5c8475186c909e9b/raw/index.html">pretty log </a></li>
<li>python38Packages.shap: <a href="https://termbin.com/q87an">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/d955ec4d5ae05e846e37963cd655139a/raw/index.html">pretty log </a></li>
<li>python39Packages.awkward: log was empty</li>
<li>python39Packages.clifford: log was empty</li>
<li>python39Packages.datashader: log was empty</li>
<li>python39Packages.pynndescent: log was empty</li>
<li>python39Packages.shap: log was empty</li>
<li>python39Packages.stumpy: log was empty</li>
  </ul>
</details>
<details>

  <summary>13 packages failed to build and are new build failures:</summary>
  <ul>
<li>python39Packages.dask-ml: log was empty</li>
<li>python39Packages.fastparquet: log was empty</li>
<li>python39Packages.graspologic: log was empty</li>
<li>python39Packages.gumath: log was empty</li>
<li>python39Packages.hyppo: log was empty</li>
<li>python39Packages.librosa: log was empty</li>
<li>python39Packages.numba: <a href="https://termbin.com/1k4v">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/fe7d5e6f901517713a0e23830ff296d7/raw/index.html">pretty log </a></li>
<li>python39Packages.pygmo: log was empty</li>
<li>python39Packages.resampy: log was empty</li>
<li>python39Packages.sparse: log was empty</li>
<li>python39Packages.stytra: log was empty</li>
<li>python39Packages.tensorly: log was empty</li>
<li>python39Packages.timezonefinder: log was empty</li>
  </ul>
</details>
<details>
<summary>
41 packages built:
</summary>
<ul>
<li>python37Packages.awkward</li>
<li>python37Packages.clifford</li>
<li>python37Packages.dask-ml</li>
<li>python37Packages.datashader</li>
<li>python37Packages.fastparquet</li>
<li>python37Packages.graspologic</li>
<li>python37Packages.gumath</li>
<li>python37Packages.hyppo</li>
<li>python37Packages.librosa</li>
<li>python37Packages.llvmlite</li>
<li>python37Packages.numba</li>
<li>python37Packages.phik</li>
<li>python37Packages.pygmo</li>
<li>python37Packages.pynndescent</li>
<li>python37Packages.resampy</li>
<li>python37Packages.sparse</li>
<li>python37Packages.stumpy</li>
<li>python37Packages.stytra</li>
<li>python37Packages.tensorly</li>
<li>python37Packages.timezonefinder</li>
<li>python38Packages.awkward</li>
<li>python38Packages.clifford</li>
<li>python38Packages.dask-ml</li>
<li>python38Packages.datashader</li>
<li>python38Packages.fastparquet</li>
<li>python38Packages.graspologic</li>
<li>python38Packages.gumath</li>
<li>python38Packages.hyppo</li>
<li>python38Packages.librosa</li>
<li>python38Packages.llvmlite</li>
<li>python38Packages.numba</li>
<li>python38Packages.phik</li>
<li>python38Packages.pygmo</li>
<li>python38Packages.pynndescent</li>
<li>python38Packages.resampy</li>
<li>python38Packages.sparse</li>
<li>python38Packages.stumpy</li>
<li>python38Packages.stytra</li>
<li>python38Packages.tensorly</li>
<li>python38Packages.timezonefinder</li>
<li>python39Packages.llvmlite</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.datashader:
<details>


warning: duplicate-check-inputs
Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- python3.7-pandas-1.1.5

Near pkgs/development/python-modules/datashader/default.nix:68:3:

```
   |
68 |   checkInputs = [
   |   ^
```

Near pkgs/development/python-modules/datashader/default.nix:47:3:

```
   |
47 |   propagatedBuildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md
</details>
python37Packages.llvmlite:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern '-static-libstdc++' doesn't match anything in file 'ffi/Makefile.linux'
```

Please check the offending substituteInPlace for typos or changes in source.

warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-static-libstdc++' doesn't match anything in file 'ffi/Makefile.linux'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
python37Packages.phik:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python37Packages.pygmo:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python37Packages.resampy:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/resampy/default.nix:32:3:

```
   |
32 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python37Packages.stytra:
<details>


warning: unclear-gpl
`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/stytra/default.nix:82:5:

```
   |
82 |     license = lib.licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
python37Packages.tensorly:
<details>


warning: duplicate-check-inputs
Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- python3.7-nose-1.3.7

Near pkgs/development/python-modules/tensorly/default.nix:28:3:

```
   |
28 |   checkInputs = [ pytest nose pytorch ];
   |   ^
```

Near pkgs/development/python-modules/tensorly/default.nix:25:3:

```
   |
25 |   propagatedBuildInputs = [ numpy scipy sparse ]
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md
</details>
python37Packages.timezonefinder:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/timezonefinder/default.nix:28:3:

```
   |
28 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python38Packages.datashader:
<details>


warning: duplicate-check-inputs
Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- python3.8-pandas-1.1.5

Near pkgs/development/python-modules/datashader/default.nix:68:3:

```
   |
68 |   checkInputs = [
   |   ^
```

Near pkgs/development/python-modules/datashader/default.nix:47:3:

```
   |
47 |   propagatedBuildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md
</details>
python38Packages.llvmlite:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern '-static-libstdc++' doesn't match anything in file 'ffi/Makefile.linux'
```

Please check the offending substituteInPlace for typos or changes in source.

warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-static-libstdc++' doesn't match anything in file 'ffi/Makefile.linux'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
python38Packages.phik:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.pygmo:
<details>


warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.resampy:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/resampy/default.nix:32:3:

```
   |
32 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python38Packages.stytra:
<details>


warning: unclear-gpl
`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/stytra/default.nix:82:5:

```
   |
82 |     license = lib.licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
python38Packages.tensorly:
<details>


warning: duplicate-check-inputs
Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- python3.8-nose-1.3.7

Near pkgs/development/python-modules/tensorly/default.nix:28:3:

```
   |
28 |   checkInputs = [ pytest nose pytorch ];
   |   ^
```

Near pkgs/development/python-modules/tensorly/default.nix:25:3:

```
   |
25 |   propagatedBuildInputs = [ numpy scipy sparse ]
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md
</details>
python38Packages.timezonefinder:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/timezonefinder/default.nix:28:3:

```
   |
28 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python39Packages.datashader:
<details>


</details>
python39Packages.llvmlite:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern '-static-libstdc++' doesn't match anything in file 'ffi/Makefile.linux'
```

Please check the offending substituteInPlace for typos or changes in source.

warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-static-libstdc++' doesn't match anything in file 'ffi/Makefile.linux'
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
</details>
python39Packages.pygmo:
<details>


</details>
python39Packages.resampy:
<details>


</details>
python39Packages.stytra:
<details>


</details>
python39Packages.tensorly:
<details>


</details>
python39Packages.timezonefinder:
<details>


</details>
</details>

veprbl committed:
python3Packages.numba: disable on python 3.9

  RuntimeError: Cannot install on Python version 3.9.2; only versions >=3.6,<3.9 are supported.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114572` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
8 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.arviz</li>
<li>python37Packages.dftfit</li>
<li>python37Packages.pygbm</li>
<li>python37Packages.pymc3</li>
<li>python38Packages.arviz</li>
<li>python38Packages.dftfit</li>
<li>python38Packages.pygbm</li>
<li>python38Packages.pymc3</li>
</ul>
</details>
<details>

  <summary>2 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python37Packages.shap: log was empty</li>
<li>python38Packages.shap: log was empty</li>
  </ul>
</details>
<details>
<summary>
41 packages built:
</summary>
<ul>
<li>python37Packages.awkward</li>
<li>python37Packages.clifford</li>
<li>python37Packages.dask-ml</li>
<li>python37Packages.datashader</li>
<li>python37Packages.fastparquet</li>
<li>python37Packages.graspologic</li>
<li>python37Packages.gumath</li>
<li>python37Packages.hyppo</li>
<li>python37Packages.librosa</li>
<li>python37Packages.llvmlite</li>
<li>python37Packages.numba</li>
<li>python37Packages.phik</li>
<li>python37Packages.pygmo</li>
<li>python37Packages.pynndescent</li>
<li>python37Packages.resampy</li>
<li>python37Packages.sparse</li>
<li>python37Packages.stumpy</li>
<li>python37Packages.stytra</li>
<li>python37Packages.tensorly</li>
<li>python37Packages.timezonefinder</li>
<li>python38Packages.awkward</li>
<li>python38Packages.clifford</li>
<li>python38Packages.dask-ml</li>
<li>python38Packages.datashader</li>
<li>python38Packages.fastparquet</li>
<li>python38Packages.graspologic</li>
<li>python38Packages.gumath</li>
<li>python38Packages.hyppo</li>
<li>python38Packages.librosa</li>
<li>python38Packages.llvmlite</li>
<li>python38Packages.numba</li>
<li>python38Packages.phik</li>
<li>python38Packages.pygmo</li>
<li>python38Packages.pynndescent</li>
<li>python38Packages.resampy</li>
<li>python38Packages.sparse</li>
<li>python38Packages.stumpy</li>
<li>python38Packages.stytra</li>
<li>python38Packages.tensorly</li>
<li>python38Packages.timezonefinder</li>
<li>python39Packages.llvmlite</li>
</ul>
</details>
